### ISSUE TITLE
look up TypeScript parameter decorators in correct scope

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1124]
```typescript
function q(...args) {}
class C {
  // Should error because param1 isn't declared.
  f(param1, @param1 param2) {}
  // Should not error because q is declared and this is not a use before declaration.
  g(@q a, q) {}
}
```
